standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -404,8 +407,7 @@ public static long updateUsingNotificationEvents(RawStore rawStore, long lastEve
   /**
    * This initializes the caches in SharedCache by getting the objects from Metastore DB via
    * ObjectStore and populating the respective caches
-   */
-  static void prewarm(RawStore rawStore) {
+   */ static void prewarm(RawStore rawStore) {
     if (isCachePrewarmed.get()) {
```

Comment:
Wonder if we shall stop loading more tables if cache is full during prewarm. Otherwise, it´s guarantee to load all tables and evict earlier tables, which seems unnecessary. But we can do it in a followup ticket.

Reply:
That's a good idea. I think I can handle that in a follow up ticket. The way we keep track of total memory size with LRU has changed, so that we don't immediately know whether the cache is full. Old code used to keep the current size and total size, and compare the two. I removed these two, since LRU tracks weights internally. To stop populating further in prewarm, I think I need to pass in the prewarm flag to SharedCache, and then in RemovalListener to check if we are in prewarm, we stop population. Since it's async, it can be a bit convoluted. Let me think it over and handle in a separate ticket. Track it here. https://issues.apache.org/jira/browse/HIVE-21855